### ISSUE TITLE
Use ntdsutil in 2008 and above to extract ntds.dit

### ISSUE THREAD

**Issue by todb-r7** [rapid7/metasploit-framework#3854]
_This issue was [RM8837](https://dev.metasploit.com/redmine/issues/8837), originally filed by Rich Rumble_

This command will extract the NTDS.dit, SYSTEM and SECURITY files all in one command, and no need to invoke shadow copy. (Although I'm sure it's doing it for you in the bg)

```
ntdsutil.exe "activate instance ntds" "ifm" "Create Full C:\Temp\123" quit quit
```

I have screen shots that show it in action on my blog and it's built-in to 2008 and greater already!

https://xinn.org/blog/JtR-AD-Password-Auditing.html

You can shorten most of the commands as well http://technet.microsoft.com/en-us/library/cc753343.aspx

And you can add "nodefrag" http://technet.microsoft.com/en-us/library/cc732530.aspx to speed up the dump a bit :)

**Comment by todb-r7** [comment]
Yes, someone should write this post module. :)

**Comment by dmaloney-r7** [comment]
@todb-r7 yeah ntdsutil is great, but unfortunately only works in newer Server OSes. It actually invokves shadowcopy for you on the backend, and pulls the exact things you need to replicate the DC. As an added benefit, it actually properly disconnects the ESEDB in ntds.dit so that it can be properly accessed offline, whereas just doing shadowcopy leaves ntds.dit in a bad state that will require additional work to repair.

**Comment by todb-r7** [comment]
Given the examples and docs referenced on the PR, seems like it'd be straightforward to implement, thus the `newbie friendly` tag.

**Comment by dmaloney-r7** [comment]
@todb-r7 my concern is that doing this we need some sort of paradigm in place for post modules to declare version compatibility, so people do not try to use it on a 2003 server. Something we'd have to probably deal with eventually anyways,

**Comment by todb** [comment]
I know today there are lots of post modules that do some version checking.
It's pretty much up to the post module in question, since there are lots of
things to look for. `grep -ril verison modules/post/` is pretty accurate.

On Thu, Mar 5, 2015 at 11:10 AM, dmaloney-r7 <notifications@github.com>
wrote:

> @todb-r7 <https://github.com/todb-r7> my concern is that doing this we
> need some sort of paradigm in place for post modules to declare version
> compatibility, so people do not try to use it on a 2003 server. Something
> we'd have to probably deal with eventually anyways,
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rapid7/metasploit-framework/issues/3854#issuecomment-77405197>
> .
>



-- 
"Tod Beardsley" <todb@packetfu.com> | 512-438-9165 | @todb
   Such coin, plz send: D98vwYv9RC4rYFzhzz3BBbrkry9bJ1GRi4

**Comment by dmaloney-r7** [comment]
@todb Yeah, but I'd like to see us add a paradigm for Metadata on the actual post module that allow the users to know ahead of time whether they should expect a post module to work on a given machine.

**Comment by egypt** [comment]
@dmaloney-r7 Was this fixed by #5348?
